src/cacheHttpClient.ts

```diff
@@ -52,39 +56,44 @@ function createAcceptHeader(type: string, apiVersion: string): string {
 
 function getRequestOptions(): IRequestOptions {
     const requestOptions: IRequestOptions = {
-        acceptHeader: createAcceptHeader("application/json", "6.0-preview.1")
+        headers: {
+            Accept: createAcceptHeader("application/json", "6.0-preview.1")
+        }
     };
 
     return requestOptions;
 }
 
-function createRestClient(): RestClient {
+function createHttpClient(): HttpClient {
     const token = process.env["ACTIONS_RUNTIME_TOKEN"] || "";
     const bearerCredentialHandler = new BearerCredentialHandler(token);
 
-    return new RestClient("actions/cache", getCacheApiUrl(), [
-        bearerCredentialHandler
-    ]);
+    return new HttpClient(
+        "actions/cache",
+        [bearerCredentialHandler],
+        getRequestOptions()
+    );
 }
 
 export async function getCacheEntry(
     keys: string[]
 ): Promise<ArtifactCacheEntry | null> {
-    const restClient = createRestClient();
+    const httpClient = createHttpClient();
     const resource = `cache?keys=${encodeURIComponent(keys.join(","))}`;
 
-    const response = await restClient.get<ArtifactCacheEntry>(
-        resource,
-        getRequestOptions()
-    );
-    if (response.statusCode === 204) {
+    const response = await httpClient.get(getCacheApiUrl(resource));
```

Comment:
I want to add httpclient.getJson<T>().  I will do that now.

Reply:
e.g. I want to make sure this code happens which would have with the rest client.  https://github.com/microsoft/typed-rest-client/blob/master/lib/RestClient.ts#L198